### DISCUSSION TITLE
[A-L] (10 settembre 2024) Esercizio 2

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/progettazione-di-algoritmi#115] [Esami A-L]
Dato un array A[] di lunghezza n di interi da 1 a n, un valore x è isolato in A se

- x è nell'array, cioè esiste un indice i tale che A[ i ] = x

- ne x - 1 ne x + 1 sono presenti nell'array, cioè se per ogni indice j, $\Large 1 \leq j \leq n,\ A[j] \neq x -1,\ A[j] \neq x + 1$.

- "1" è isolato in A se 1 è presente in A, ma 2 non è presente in A. "n" è isolato in A se n è presente in A ma n - 1 non è presente.

Dare lo pseudocodice di un algoritmo che prende in input L'array A e in complessità O(n) trova tutti i valori isolati di A.

**Comment by daromacarlo** [comment]
![immagine](https://github.com/user-attachments/assets/8790871b-5968-432d-aca9-d3bffbe6251d)

**Comment by niccolo-fato** [comment]
![Screenshot 2025-05-24 alle 21 16 18](https://github.com/user-attachments/assets/3829936f-a048-44dd-872c-3ad77d006c7c)

**Reply by Near047** [reply]
Scusate in questo esercizio però non capisco come sia formata A, è una lista di interi da 1 a n, quindi il massimo è n, perciò l'unico modo è che ci sia qualche ripetizione di interi?

**Comment by cek753** [comment]
Per questo algoitmo uso un array ausiliario di lunghezza n per tenere conto dei numeri presenti in A(che vanno da 1 a n)

```
algo(A):
    sol={}
    s=[0]*n
    n = len(A)
    for i in range(A)
        s[A[i]]=1
    for x in len(A):
        if A[x] ==1:
             if s[2]==0: sol.add{1}
        elif x == n:
             if (s[A[n]-1])==0: sol.add{A[n]}
        elif (s[A[x]-1])==0 and (s[A[x]+1])==0: sol.add{A[x]}
        
   return sol
```
